M3-2468 Add: Theme Spacing/Compact mode Switcher

alioso opened the pull request:
## Theme Spacing/Compact mode Switcher

This PR is a big theme rework to allow more user preference values and customization. Along with the light/dark theme, we feature now a switch to select a compact/normal mode.

There was quite a bit of refactoring happen due to renderGuard and exports, and there was a lof of styles that for fixed. This work also will pave the way for any other customization we'd like to implement in the future.

## Type of Change
- Non breaking change ('update', 'change')

## Note to Reviewers

The compact theme should be tested everywhere to check for consistency. I also recommend switching back and forth through the testing to check for performance hits (lots of re-rendering can happen). I have not noticed anything wrong, and users won't toggle like crazy, but something to keep in mind.

WilkinsKa1 commented on src/themeFactory.ts:
```diff
@@ -91,1078 +97,1091 @@ const iconCircleAnimation = {
+      MuiDialogActions: {
+        root: {
+          margin: 0,
+          padding: `0 ${spacingUnit * 3}px, ${spacingUnit *
```
Need to remove the commas here, otherwise causes regression ( I would but for whatever reason I could only checkout your branch in a detached state 😞):
<img width="432" alt="screen shot 2019-02-28 at 4 47 16 pm" src="https://user-images.githubusercontent.com/2565527/53601715-b75ec100-3b7a-11e9-870c-cc15231b5d83.png">

alioso replied:
@WilkinsKa1 Addressed, good catch!

alioso commented:
@johnwcallahan thx for the thorough feedback! I updated all instances of updateFor to include classes and that fixed all the styles not re-rendering in the create workflow. 

I also fixed the back up panel and the loading button

alioso commented:
@johnwcallahan addressed as well!
